src/rpc/handlers/LedgerEntry.cpp

```diff
@@ -157,10 +157,27 @@ LedgerEntryHandler::process(LedgerEntryHandler::Input input, Context const& ctx)
         return Error{*status};
 
     auto const lgrInfo = std::get<ripple::LedgerHeader>(lgrInfoOrStatus);
-    auto const ledgerObject = sharedPtrBackend_->fetchLedgerObject(key, lgrInfo.seq, ctx.yield);
-
-    if (!ledgerObject || ledgerObject->empty())
-        return Error{Status{"entryNotFound"}};
+    std::uint32_t deleted_ledger_index = 0;
+    bool ledger_deleted = false;
+    std::optional<data::Blob> ledgerObject;
+    if (include_deleted){
+        auto const lastTwoObjects = sharedPtrBackend_->fetchLastTwoLedgerObjects(key, lgrInfo.seq, ctx.yield);
+        if (lastTwoObjects.empty())
+            return Error{Status{"entryNotFound"}};
+        if (lastTwoObjects.size() == 2) {
+            ledgerObject = lastTwoObjects[0].second.empty()? std::make_optional(lastTwoObjects[1].second) : std::make_optional(lastTwoObjects[0].second);
+            deleted_ledger_index = lastTwoObjects[0].second.empty()? lastTwoObjects[0].first : 0;
+            ledger_deleted = true;
+        } else {
```

Comment:
should group the statements as follows, for readability

```
       if (lastTwoObjects.empty()){...}
       else if (lastTwoObjects.size() == 1) {
           ....
        }
        else if (lastTwoObjects.size() == 2) {
             ...
        }
```

 we should avoid using `else`  clause because other people could assume that the size of the `lastTwoObjects` can be greater than 2, which leads to confusion